goon/browserassets/js/browserOutput.js

```diff
@@ -171,25 +171,67 @@ function setHighlightColor(match) {
 	match.style.background = opts.highlightColor
 }
 
+function escapeRegex(input){ // put this in a function incase it ever needs to be used elsewhere, makes code cleaner
+	return input.replace(/[.*+?^${}()|[\]\\]/g, '\\$&');
+}
+
 //Highlights words based on user settings
 function highlightTerms(el) {
 	if(regexHasError) return; //just stop right there ig the regex is gonna except
+	function highlightRecursor(element,term){ //recursor function to do the highlighting proper
+		var regex = new RegExp("("+term+")","gi")
+		function replace(str) {
+			return str.replace(regex,'<span style="background-color:'+opts.highlightColor+'">$&</span>')
+		}
+		var s=''
+		var work=element.innerHTML;
+		var work_lc=work.toLowerCase()
+		var ind=0;
+		while(ind < work_lc.length) {
+			console.log(s);
+			var next_term=work_lc.substring(ind).search(regex);
+            if(next_term!=-1)next_term+=ind;
```

Comment:
indentation happens, but how the fuck and why are braceless ifs discouraged

Reply:
Here's how: https://www.imperialviolet.org/2014/02/22/applebug.html

Also see this stackoverflow question (though it's mostly the same thing): https://stackoverflow.com/questions/2125066/is-it-a-bad-practice-to-use-an-if-statement-without-curly-braces

Reply:
Dont bring apple into this, we all know they're idiots already.
the SO question was closed for opinionated, the only reason it would be bad would be people being absolute idiots, which isnt a reason to not use it.